dds/DCPS/DomainParticipantImpl.cpp

```diff
@@ -1251,9 +1296,11 @@ DomainParticipantImpl::ignore_topic(
   if (!disco->ignore_topic(domain_id_,
                            dp_id_,
                            ignoreId)) {
-    ACE_ERROR((LM_ERROR,
-               ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::ignore_topic, ")
-               ACE_TEXT(" Could not ignore topic.\n")));
+    if (DCPS_debug_level > 0) {
+      ACE_ERROR((LM_ERROR,
+                ACE_TEXT("(%P|%t) ERROR: DomainParticipantImpl::ignore_topic, ")
+                ACE_TEXT(" Could not ignore topic.\n")));
+    }
```

Comment:
I think the convention is that it's unconditional if the error doesn't have an error return. Is that right @mitza-oci?

Reply:
As user I would expect OpenDDS to be silent with logging when there is no logging level set, in order to work around some OpenDDS issues we for example call some operations which could fail because for example some other module already has created the topic, at that moment we take a different path, same in cleanup

Reply:
Do we want this case to be `RETCODE_OK`?  If it returned an error it would probably have no need to log (the discovery implementation can log any internal problems).

Reply:
Not sure, maybe something to test and change separately?

Reply:
Looks I misinterpreted your first comment, now see what you exactly mean

Reply:
Shouldn't at the moment the ignore_topic calls on discovery fails the addition to ignored_topics_ be reverted?

Reply:
Created new issue #2079 for this specific line